src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -519,15 +519,25 @@ private boolean checkForShovel()
         return needsShovel;
     }
 
-    private boolean checkForTool(@NotNull final String tool)
+     /**
+     * Ensures that we have a tool available.
+     *
+     * @param tool tool required for block
+     * @return true if we have a tool
+     */
+    private boolean checkForTool(@NotNull String tool)
     {
         final boolean needsTool = !InventoryFunctions
                                      .matchFirstInInventory(
                                        worker.getInventoryCitizen(),
                                        stack -> Utils.isTool(stack, tool),
                                        InventoryFunctions::doNothing
                                      );
-        if (!needsTool)
+
+        final boolean isUsable = hasToolLevel(tool);
+        final int hutLevel = worker.getWorkBuilding().getBuildingLevel();
+
+        if (!needsTool && isUsable)
```

Comment:
why &&
if he does not need a tool, break
if the tool he got is fine, break

If I am not mistaken, a || will work and allow you to inline isUseable